library/include/rocwmma/internal/layout_impl.hpp

```diff
@@ -851,23 +848,26 @@ namespace rocwmma
                 {
                     using OrthoLayout
                         = ColInlineVW<BlockDim, BlockK, DataT, VectorWidth, MaxVectorWidth>;
-                    using MatrixCoordT = std::pair<int32_t, int32_t>;
+                    using MatrixCoordT = HIP_vector_type<uint32_t, 2>;
                 };
 
                 // Matrix coord offsets
                 __device__ static inline typename Traits::MatrixCoordT baseOffset()
                 {
-                    return std::swap(Traits::OrthoLayout::baseOffset());
+                    return Coord2d(Traits::OrthoLayout::baseOffset().y,
```

Comment:
Swap utility will also avoid the double 'rhs' value generation of baseOffset(), once for x and once for y. baseOffset returns by value.